Updated principal-key/features/functions.md based on AA feedback

Andriciuc opened the pull request:
In set-principal-key.md:
* updated with correct code example using set_server_key_using_global parameter
* updated note to reflect correct config

In features.md:
* Removed temporary tables feature to clear confusion, removed logical replication mention, removed WAL encryption as a feature.

In functions.md:
* Added ON FUNCTION for grant/revoke execution
* Modified sensitive info bolded paragraph to important note
* Small modifications to notes display, title cases and text fixes
* added note to Add or modify Vault providers for keeping the same principal key.
* Added warning for WAL in pg_tde_create_key_using_global_key_provider

In general:
* Removed all logical replication mentions except the FAQ and in RC2 release note.

Andriciuc committed:
initial update from AA feedback

In set-principal-key.md:
* updated with correct code example using set_server_key_using_global parameter
* updated note to reflect correct config

In features.md:
* Removed temporary tables feature to clear confusion, removed logical replication mention, removed WAL encryption as a feature.

In functions.md:
* Added ON FUNCTION for grant/revoke execution
* Modified sensitive info bolded paragraph to important note
* Small modifications to notes display, title cases and text fixes
* added note to Add or modify Vault providers for keeping the same principal key.
* Added warning for WAL in pg_tde_create_key_using_global_key_provider

In general:
* Removed all logical replication mentions except the FAQ and in RC2 release note.

AndersAstrand commented on contrib/pg_tde/documentation/docs/functions.md:
```diff
@@ -225,14 +230,14 @@ These functions list the details of all key providers for the current database o
 
-!!! important
+!!! warning
     All configuration values include possibly sensitive values, such as passwords. **Never** specify these directly, use the remote configuration option instead.
```
We don't have the remote option any more. It's all file paths now.

It's no longer possible to specify secrets directly

Andriciuc replied:
would this work? Never specify these directly, use the file paths instead.

AndersAstrand replied:
Sorry, I edited my original comment before I saw yours because I realized it was unclear. We no longer allow sensitive values to be specified directly, they are all file paths now.

Andriciuc replied:
I got it! New warning: 

!!! warning
    Do **not** specify sensitive values, such as passwords, directly in the configuration. All sensitive parameters must now be provided via file paths.

AndersAstrand replied:
It's not possible to do it directly. so the warning can just be removed :)

Andriciuc replied:
Oooh, I thought we could, thank you Andres, makes sense, removed. :)

Andriciuc committed:
Update functions.md

* fixed keyfile to key file and keyfile proviers to key file providers
* updated remote config options to file paths for specific param warning note.

AndersAstrand commented on contrib/pg_tde/documentation/docs/features.md:
```diff
@@ -9,16 +9,13 @@ The following features are available for the extension:
     * Data tables
     * Index data for encrypted tables
     * TOAST tables
-    * Temporary tables created during database operations
```
It's fine to leave "Temporary tables" in here since we do encrypt those. It's just that everything here is "created during database operations" so I didn't understand that part :)

Andriciuc replied:
Initially I wanted to write: Temporary tables created during queries on encrypted tables are also encrypted

I am thinking that perhaps this is just a standard PostgreSQL feature, so maybe I should remove it, thoughts?

AndersAstrand replied:
Temporary tables is a standard feature that we can encrypt. You're thinking about temporary files created during query execution. those are not (yet) encrypted.

Andriciuc replied:
Gotcha, thank you Anders! Left the following:

* Temporary tables

Andriciuc committed:
Update functions.md

fixed keyfile to key file for title

Removed warning notes regarding remote config options.

Andriciuc committed:
Update functions.md

restructured and improved paragraphs create key and creates keys

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Update features.md

Added * Temporary tables

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

AndersAstrand reviewed (approved):
Text looks good, please squash before commit.

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey

Andriciuc committed:
Merge branch 'release-17.5.2' into fixes-for-features-functions-setprincikey
